web/src/components/graph/areagraph.js

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from 'react';
+import { scaleTime, scaleLinear } from 'd3-scale';
+import { first, last } from 'lodash';
+import moment from 'moment';
+
+import AreaGraphGradients from './areagraphgradients';
+import AreaGraphLayers from './areagraphlayers';
+import GraphBackground from './graphbackground';
+import GraphHoverLine from './graphhoverline';
+import ValueAxis from './valueaxis';
+import TimeAxis from './timeaxis';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 35;
+const Y_AXIS_PADDING = 4;
+
+const getTimeScale = (containerWidth, datetimes, currentTime) => scaleTime()
+  .domain([first(datetimes), currentTime ? moment(currentTime).toDate() : last(datetimes)])
+  .range([0, containerWidth]);
+
+const getValueScale = (containerHeight, maxTotalValue) => scaleLinear()
+  .domain([0, maxTotalValue * 1.1])
+  .range([containerHeight, Y_AXIS_PADDING]);
+
+const AreaGraph = React.memo(({
+  id,
+  layers,
+  currentTime,
```

Comment:
is this a `Date` object? If it is, maybe we should call it currentDate?

Reply:
I must confess I don't fully understand which values `currentTime` can assume and how it's being used, but to answer your questions, it looks like a `DateTime` object and it _can't_ be inferred from `selectedTimeIndex`.

[This](https://github.com/tmrowco/electricitymap-contrib/pull/2171/files#diff-d6c919ee9d0f0d9969d62718df324219R125) is its definition currently in use:

```
state.application.customDate || (state.data.grid || {}).datetime
```

So overall it seems there are currently 3 ways the end of the `AreaGraph` time scale interval can be set (in order):

1. By directly setting a `customDate` value
2. From `data.grid.datetime`
3. From the last timestamp from the data API response

I'm not really sure what the use cases for (1) and (2) are - if you think they are just legacy, then we can get rid of `currentTime` as a prop altogether and use a max between (3) and the actual current time for setting the time scale end.

Reply:
Actually it seems like `currentTime` is used here:
https://github.com/tmrowco/electricitymap-contrib/pull/2171/files#diff-e666df2f408c2db2102da3e7af69650aR28-R30
It's used to set the scale. The reasoning is this:
If the history data hasn't been updated in a while, the last datapoint will have a datetime that is much older than now(). Therefore, we pass the `currentTime` (which represents now()) in order to force the scale to go up to now().

There's two things we can do:
1. remove the prop and that logic in AreaGraph
2. rename the prop to make it less confusing (something like maxDatetime could be better)

I suggest going with 2. and make it optional.

Reply:
Indeed, so what I was suggesting above is we go with (1) but instead I went for a bit more robust version of (2) in https://github.com/tmrowco/electricitymap-contrib/pull/2171/commits/2f4c419dbc9d66fe5f756cc9d5bef185f395056f.

Actually, the way I currently understand it, the same reasoning you provided could be applied to the other end of the graph to argue we need `startTime` to show gaps when data is missing at the beginning of graph, so I added support for it, but didn't fully transition as the other graphs are not using it yet and for the UI consistency, they should probably be switched at the same time.

I opened https://github.com/tmrowco/electricitymap-contrib/issues/2250 to capture the essence of the problem - please let me know if that makes sense.

Reply:
That makes sense!